x-pack/plugins/synthetics/server/routes/settings/private_locations/delete_private_location.ts

```diff
@@ -22,19 +24,42 @@ export const deletePrivateLocationRoute: SyntheticsRestApiRouteFactory<
 > = () => ({
   method: 'DELETE',
   path: SYNTHETICS_API_URLS.PRIVATE_LOCATIONS + '/{locationId}',
-  validate: {
-    params: schema.object({
-      locationId: schema.string({ minLength: 1, maxLength: 1024 }),
-    }),
+  validate: {},
+  validation: {
+    request: {
+      params: schema.object({
+        locationId: schema.string({ minLength: 1, maxLength: 1024 }),
+      }),
+    },
   },
   writeAccess: true,
-  handler: async ({ savedObjectsClient, syntheticsMonitorClient, request }) => {
+  handler: async ({ response, savedObjectsClient, syntheticsMonitorClient, request }) => {
     const { locationId } = request.params as { locationId: string };
 
     const { locations, agentPolicies } = await getPrivateLocationsAndAgentPolicies(
       savedObjectsClient,
       syntheticsMonitorClient
     );
+
+    if (!locations.find((loc) => loc.id === locationId)) {
+      return response.badRequest({
+        body: {
+          message: `Private location with id ${locationId} does not exist.`,
+        },
+      });
+    }
+
+    const monitors = await getMonitorsByLocation(savedObjectsClient, locationId);
```

Comment:
There's an edge case that a Private Location attached to a monitor defined in `Default` space can be deleted from a different space (e.g. `s2`)

```console
➜  ~ curl -X DELETE "http://localhost:5602/omz/api/synthetics/private_locations/elastic-agent-managed-ep" -H "Elastic-API-Version:2023-10-31" -u elastic:changeme -H "kbn-xsrf: true"
// {"statusCode":400,"error":"Bad Request","message":"Private location with id elastic-agent-managed-ep cannot be deleted because it is used by 1 monitor(s)."}%

➜  ~ curl -X DELETE "http://localhost:5602/omz/s/s2/api/synthetics/private_locations/elastic-agent-managed-ep" -H "Elastic-API-Version:2023-10-31" -u elastic:changeme -H "kbn-xsrf: true"
// # []% 
```

In the second attempt above, Private Location `elastic-agent-managed-ep` has been deleted leaving the monitor in an invalid state.